master/internal/task/allocation.go

```diff
@@ -356,13 +351,13 @@ func (a *Allocation) Cleanup(ctx *actor.Context) {
 // heavy stuff unless it is necessarily (which also works to spread occurrences of the same work
 // out). Eventually, Allocations should just be started with their TaskSpec.
 func (a *Allocation) ResourcesAllocated(ctx *actor.Context, msg sproto.ResourcesAllocated) error {
-	if a.state != model.AllocationStatePending {
+	if a.getModelState() != model.AllocationStatePending {
 		// If we have moved on from the pending state, these must be stale (and we must have
 		// already released them, just the scheduler hasn't gotten word yet).
 		return ErrStaleResourcesReceived{}
 	}
 
-	a.state = model.AllocationStateAssigned
+	a.setModelState(model.AllocationStateAssigned)
```

Comment:
ideally, at this point it would get automagically persisted or something (the entire model) at the end of a receive block. could `defer persist model` at the top of receive and use a flag on the model to decide to persist. also, eventually bun.

Reply:
maybe ideally is too strong. "i had a thought that, " instead.

Reply:
yes I agree we could do that one day.
also some of the methods here do two updates within one function without a transaction, which also would be a nice improvement once we have bun here.